### ISSUE TITLE
Extract NAPALM integrations to an app/plugin

### ISSUE THREAD

**Issue by glennmatthews** [nautobot/nautobot#2557]
### Proposed Changes

Extract the existing integrations with NAPALM to a separate app/plugin. This would include:

- Device tabs (LLDP neighbors, configuration) - now that plugins can provide custom tabs this should be replaceable.
- REST API endpoint for making NAPALM requests (plugin can provide this)
- NAPALM provider and NAPALM args as core fields on `Platform` (plugin can define custom fields or a custom data model)

### Justification

Streamlining, simplification, improving modularity of core.

Relates to #1008 as a feature request.

**Comment by jvanderaa** [comment]
Yes please! Long overdue to have this included. And benefit is that a plugin may be able to take things even further than just the three specific tabs there are now.

**Comment by itdependsnetworks** [comment]
Can someone confirm, is the napalm capabilities blocking or not?

**Comment by lampwins** [comment]
> Can someone confirm, is the napalm capabilities blocking or not?

@itdependsnetworks if you mean blocking execution, then yes.

**Comment by itdependsnetworks** [comment]
Thanks. Is it safe to say in a new plugin, this should/would not be the case?

**Comment by glennmatthews** [comment]
If I understand where you're coming from correctly, based on our conversation this morning, you're thinking that in a plugin, NAPALM calls should all execute asynchronously in a background task (a la #525), rather than synchronously as in the current implementation. Definitely worth considering, although we need to think about results storage (as mentioned in https://github.com/nautobot/nautobot/issues/525#issuecomment-853362684).

**Comment by itdependsnetworks** [comment]
egg-zactly
